### PR TITLE
Bfs Implementation

### PR DESCRIPTION
**Pull request by rohitvarkey**

### CODE REVIEW

File: src/algorithms/bfs.jl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        vertexneighbors = getsuccessors(s, src)
+        for vertex in vertexneighbors
+            #If not already set, and is not found in the queue.
+            if !visited[vertex]
+                push!(next, vertex) #Push onto queue
+                visited[vertex] = true #Mark that it is has been visited.
+            end
+        end
+    end
+    findin(visited, true) #Return all vertices that are set to true
```
Comment (jpfairbanks) [inline src/algorithms/bfs.jl]: you should return tree stored in the form of a parents array where parent[i] is the vertex name of the vertex that pointed to i.

File: src/algorithms/bfs.jl [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    next = Vector{Int64}([source])
+    sizehint!(next, s[max_nv])
+    while !isempty(next)
+        src = shift!(next) #Get first element
+        vertexneighbors = getsuccessors(s, src)
+        for vertex in vertexneighbors
+            #If not already set, and is not found in the queue.
+            if !visited[vertex]
+                push!(next, vertex) #Push onto queue
+                visited[vertex] = true #Mark that it is has been visited.
```
Comment (jpfairbanks) [inline src/algorithms/bfs.jl]: parent[next] = vertex

**Comment by rohitvarkey** [comment]
@jpfairbanks What should the value of the source vertex be in the parents array returned? I've put that to `-1` currently to signify no parent.

**Comment by jpfairbanks** [comment]
Yeah -1 for source and -2 for no parents is fine. Its just a convention that we can change later if there is a good argument to do something else.

**Pull request merged**